src/com/ucm/dasi/catan/game/generator/CatanDeterministicGenerator.java

```diff
@@ -0,0 +1,99 @@
+package com.ucm.dasi.catan.game.generator;
+
+import java.util.Random;
+
+public class CatanDeterministicGenerator extends CatanGenerator
+    implements IDeterministicNumberGenerator {
+
+  private static final int N_0 = 0;
+  private static final int N_1 = 0;
+  private static final int N_2 = 1;
+  private static final int N_3 = 2;
+  private static final int N_4 = 3;
+  private static final int N_5 = 4;
+  private static final int N_6 = 5;
+  private static final int N_7 = 6;
+  private static final int N_8 = 5;
+  private static final int N_9 = 4;
+  private static final int N_10 = 3;
+  private static final int N_11 = 2;
+  private static final int N_12 = 1;
+
+  private int innerOffset;
+
+  private int[] productionSequence;
+
+  public CatanDeterministicGenerator() {
+
+    innerOffset = 0;
+    productionSequence = generateSequence();
+  }
+
+  @Override
+  public int getNextProductionNumber() {
+
+    int productionNumber = getProductionNumber(0);
+
+    innerOffset = (innerOffset + 1) % productionSequence.length;
+
+    return productionNumber;
+  }
+
+  @Override
+  public int getProductionNumber(int offset) {
+
+    int index = (offset + innerOffset) % productionSequence.length;
+    if (index < 0) {
+      index += productionSequence.length;
+    }
+
+    return productionSequence[index];
+  }
+
+  private int[] generateSequence(int[] remainingNumbersSet) {
+
+    int sum = 0;
+
+    for (int numberAmount : remainingNumbersSet) {
```

Comment:
Maybe you can create an static array of the numbers insted of passing throught an argument.